File: mario.cs
// approximate refactor of "Mario" problem in C#

using cs50.h;
using System; 

namespace Mario
{
    class Program
    {
        static void Main(string[] args)
        {
            do
            {
            Console.WriteLine("Enter pyramid height: ");
            int height = Console.ReadLine();
            }
            while (height < 0 || height > 23);

            for (int i = 1; i <= height; i++)
            {
                for (int j = 0; j < (height - i); j++)
                {
                   Console.WriteLine(" ");
                }

                for (int k = 0; k < i + 1; k++)
                {
                    Console.WriteLine("#");
                }

                Environment.NewLine;
            }
        }
    }
}

File: mario.c
// CS50 Problem Set 1 "Mario"

#include <cs50.h> //proprietary CS50 libraries
#include <stdio.h>

int main(void)
{
    int height, i, j, k;
    
    // get pyramid height
    do
    {
        printf("Enter your pyramid height: ");
        height = GetInt();
    }
    while (height < 0 || height > 23);
    
    // print lines (vertical)
    for (i = 1; i <= height; i++)
    {
        // print spaces (horizontal)
        for (j = 0; j < (height - i); j++)
        {
            printf(" ");
        }
        
        // print hashes (horizontal)
        for (k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        
        printf("\n");
    }
    return 0; //success! 
}